client/main.js

```diff
@@ -0,0 +1,470 @@
+import BN from "bn.js";
+import web3Utils from "web3-utils";
+
+const ganache = require("ganache-core");
+const contract = require("truffle-contract");
+const ReputationMiningCycleJSON = require("../build/contracts/ReputationMiningCycle.json");
+const ColonyNetworkJSON = require("../build/contracts/IColonyNetwork.json");
+const PatriciaTreeJSON = require("../build/contracts/PatriciaTree.json");
+
+const ReputationMiningCycle = contract(ReputationMiningCycleJSON);
+
+const jsonfile = require("jsonfile");
+
+const file = "./reputations.json";
+
+const Web3 = require("web3");
+
+const ColonyNetwork = contract(ColonyNetworkJSON);
+const accountAddress = "0xbb46703786c2049d4d6dd43f5b4edf52a20fefe4";
+
+class ReputationMiningClient {
```

Comment:
Just a general comment about potentially synchronising the function names between this and the rep mining client. Below I've called them out for ease of comparing
`binarySearchForChallenge` / `respondToBinarySearch`
`respondToChallenge` / `respondToChallenge`
`submitJRH` / `submitJustificationRootHash`
`submitNewHash` / `submitRootHash`

Reply:
I have gone with 
`respondToBinarySearchForChallenge`
`respondToChallenge`
`submitJustificationRootHash`
`submitRootHash`